pulsar-client/src/main/java/org/apache/pulsar/client/impl/TableViewImpl.java

```diff
@@ -289,20 +373,27 @@ private void readAllExistingMessages(Reader<T> reader, CompletableFuture<Reader<
                                reader.getTopic(),
                                messagesRead,
                                durationMillis / 1000.0);
-                       future.complete(reader);
+                       future.complete(null);
                    }
                 });
     }
 
     private void readTailMessages(Reader<T> reader) {
         reader.readNextAsync()
                 .thenAccept(msg -> {
-                    handleMessage(msg);
-                    readTailMessages(reader);
+                    try {
+                        handleMessage(msg);
+                        checkAllFreshTask(msg);
+                        readTailMessages(reader);
```

Comment:
Move `readTailMessages(reader)` out of the try block because it does not depend on the `msg`.

BTW, this PR involves many code changes to `handleMessage` because you moved the `try-finally` block out of the `handleMessage`. You can just add the logic of `checkAllFreshTask` at the end of `handleMessage` to avoid so many code changes.

```java
    private void handleMessage(Message<T> msg) {
        try {
            // ...
            checkAllFreshTask(msg);
        } finally {
            msg.release();
        }
    }
```